mne/morph.py

```diff
@@ -0,0 +1,1443 @@
+    header['pixdim'][4] = 1e3 * stc.tstep
+
+    with warnings.catch_warnings(record=True):  # nibabel<->numpy warning
+        img = NiftiImage(vol, affine, header=header)
+
+    # if a specific voxel size was targeted (only possible after morphing)
+    if voxel_size_defined:
+        # reslice mri
+        img, img_affine = reslice(
+            img.get_data(),
+            img.affine,
+            _get_zooms_orig(morph_data),
+            voxel_size)
+        with warnings.catch_warnings(record=True):  # nibabel<->numpy warning
+            img = NiftiImage(img, img_affine, header=header)
+
+    return img
+
+
+###############################################################################
+# Morph for VolSourceEstimate
+@verbose
+def _compute_morph_sdr(mri_from, mri_to,
+                       niter_affine=(100, 100, 10),
+                       niter_sdr=(5, 5, 3),
+                       spacing=(5., 5., 5.),
+                       verbose=None):
+    """Get a matrix that morphs data from one subject to another.
+
+    Parameters
+    ----------
+    mri_from : str | Nifti1Image
+        Path to source subject's anatomical MRI or Nifti1Image
+    mri_to : str | Nifti1Image
+        Path to destination subject's anatomical MRI or Nifti1Image
+    niter_affine : tuple of int
+        Number of levels (``len(niter_affine)``) and number of
+        iterations per level - for each successive stage of iterative
+        refinement - to perform the affine transform.
+        Default is niter_affine=(100, 100, 10)
+    niter_sdr : tuple of int
+        Number of levels (``len(niter_sdr)``) and number of
+        iterations per level - for each successive stage of iterative
+        refinement - to perform the Symmetric Diffeomorphic Registration (sdr)
+        transform. Default is niter_sdr=(5, 5, 3)
+    spacing : tuple | int | float | None
+        Voxel size of volume for each spatial dimension separately (tuple) or
+        isometric (int).
+        If spacing is None, MRIs won't be resliced. Note that in this case
+        both volumes must have the same number of slices in every
+        spatial dimension.
+    verbose : bool | str | int | None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+
+    Returns
+    -------
+    morph : dict
+        Details about AffineMap, DiffeomorphicMap, morph_shape, morph_zooms,
+        morph_affine for affine and diffeomorphic registration.
+
+    Notes
+    -----
+    This function will be used to morph VolSourceEstimate based on an
+    affine transformation and a nonlinear morph, estimated based on
+    respective transformation from the subject's anatomical T1 (brain) to
+    a destination subject's anatomical T1 (e.g. fsaverage). Afterwards the
+    transformation can be applied. Affine transformations are computed
+    based on the mutual information. This metric relates structural changes
```

Comment:
reword this part
```
This metric relates structural changes
    in image intensity values. Because different still brains expose high
    structural similarities this method works quite well to relate
    corresponding features [1]_.
```